src/webgpu/api/validation/capability_checks/limits/maxSampledTexturesPerShaderStage.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { range, reorder, kOrderKeys } from '../../../../../common/util/util.js';
+import { kShaderStageCombinationsWithStage } from '../../../../capability_info.js';
+
+import { kLimitBaseParams, makeLimitTestGroup } from './limit_utils.js';
+
+const limit = 'maxSampledTexturesPerShaderStage';
+export const { g, description } = makeLimitTestGroup(limit);
+
+g.test('createBindGroupLayout,at_over')
+  .desc(
+    `
+  Test using at and over ${limit} limit in createBindGroupLayout
+  
+  Note: We also test order to make sure the implementation isn't just looking
+  at just the last entry.
+  `
+  )
+  .params(
+    kLimitBaseParams
+      .combine('visibility', kShaderStageCombinationsWithStage)
+      .combine('order', kOrderKeys)
+  )
+  .fn(async t => {
+    const { limitTest, testValueName, visibility, order } = t.params;
+    await t.testDeviceWithRequestedLimits(
+      limitTest,
+      testValueName,
+      async ({ device, testValue, shouldError }) => {
+        await t.expectValidationError(() => {
+          device.createBindGroupLayout({
+            entries: reorder(
+              order,
+              range(testValue, i => ({
+                binding: i,
+                visibility,
+                texture: {},
+              }))
+            ),
+          });
+        }, shouldError);
+      }
+    );
+  });
```

Comment:
Same as the other PR, will need to test automatic layouts coming from create\*pipeline\*

Should probably be a followup, since it's going to be a bunch more work.

Reply:
I will add after the bind limits test is merged as it has the utils need to make this easier